java/email-contact-form/src/Main.java

```diff
@@ -0,0 +1,133 @@
+            String referer = context.getReq().getHeaders().get("referer");
+            String errorCode = ErrorCode.INVALID_REQUEST;
+            return context.getRes().redirect(
+                    String.format("%s?code=%s", referer, errorCode)
+            );
+        }
+
+        if(!Cors.isOriginPermitted(context)){
+            context.error("Origin not permitted");
+            String referer = context.getReq().getHeaders().get("referer");
+            String errorCode = ErrorCode.INVALID_REQUEST;
+            return context.getRes().redirect(
+                    String.format("%s?code=%s", referer, errorCode)
+            );
+        }
+
+        Map<String, List<String>> formData = new HashMap<>();
+        String body = (String) context.getReq().getBody();
+        String[] params = body.split("&");
+        for (String param : params) {
+            String[] keyValue = param.split("=");
+            String key = keyValue[0];
+            String value = keyValue[1];
+            if (!formData.containsKey(key)) {
+                formData.put(key, new ArrayList<>());
+            }
+            formData.get(key).add(value);
+        }
+
+        Map<String, String> form = new HashMap<>();
+        for (Map.Entry<String, List<String>> entry : formData.entrySet()) {
+            form.put(entry.getKey(), entry.getValue().get(0));
+        }
+
+        try {
+            Utils.throwIfMissing(form, Collections.singletonList("email"));
+        } catch (IllegalArgumentException ex) {
+            String referer = context.getReq().getHeaders().get("referer");
+            String errorCode = ErrorCode.MISSING_FORM_FIELDS;
+            return context.getRes().redirect(
+                    String.format("%s?code=%s", referer, errorCode),
+                    301,
+                    Cors.getCorsHeaders(context)
+            );
+        }
+
+
+        try {
+            Map<String, String> emailOptions = new HashMap<>();
+            emailOptions.put("from", System.getenv("SMTP_USERNAME"));
+            emailOptions.put("to", System.getenv("SUBMIT_EMAIL"));
+            emailOptions.put("subject", "New Contact Form Submission");
+            emailOptions.put("text", Utils.templateFormMessage(form));
+            Utils.sendEmail(emailOptions);
+        } catch (MessagingException ex) {
+            context.log("MessagingException: " + ex.getMessage());
+            String referer = context.getReq().getHeaders().get("referer");
+            String errorCode = ErrorCode.SERVER_ERROR;
+            return context.getRes().redirect(
+                    String.format("%s?code=%s", referer, errorCode),
+                    301,
+                    Cors.getCorsHeaders(context)
+            );
+        } catch (Exception ex) {
+            context.log("Exception: " + ex.getMessage());
+            String referer = context.getReq().getHeaders().get("referer");
+            String errorCode = ErrorCode.SERVER_ERROR;
```

Comment:
I don't think we need this variable, just move it inline

Reply:
Removed the errorCode variable and moved errorCode it to inline